nixos/modules/services/x11/redshift.nix

```diff
@@ -35,42 +46,41 @@ in {
       '';
     };
 
-    temperature = {
-      day = mkOption {
-        type = types.int;
-        default = 5500;
-        description = ''
-          Colour temperature to use during the day, between
-          <literal>1000</literal> and <literal>25000</literal> K.
-        '';
-      };
-      night = mkOption {
-        type = types.int;
-        default = 3700;
-        description = ''
-          Colour temperature to use at night, between
-          <literal>1000</literal> and <literal>25000</literal> K.
-        '';
-      };
+    settings = mkOption {
+      type = with types; attrsOf (attrsOf (nullOr (oneOf [ str float bool int ])));
+      default = {};
+      description = ''
+        The configuration to pass to redshift.
+        See <command>man redshift</command> under the section
+        CONFIGURATION FILE for options.
+      '';
+      example = literalExample ''{
+        redshift = {
+          dawn-time = "05:00-06:00";
+          dusk-time = "22:00-23:00";
+          temp-night = 3000;
+        };
+      };'';
+      apply = c:
+        if !(c ? redshift.dawn-time || c ? redshift.dusk-time) && !(c ? redshift.location-provider) && options.locations.provider.isDefined then
+          c // {
+            redshift.location-provider = lcfg.provider;
+          }
+        else
+          c;
     };
 
-    brightness = {
-      day = mkOption {
-        type = types.str;
-        default = "1";
-        description = ''
-          Screen brightness to apply during the day,
-          between <literal>0.1</literal> and <literal>1.0</literal>.
-        '';
-      };
-      night = mkOption {
-        type = types.str;
-        default = "1";
-        description = ''
-          Screen brightness to apply during the night,
-          between <literal>0.1</literal> and <literal>1.0</literal>.
-        '';
-      };
+    configFile = mkOption {
+      type = types.path;
+      example = "~/.config/redshift/redshift.conf";
+      description = ''
+        Configuration file for redshift. It is recommended to use the
+        <option>settings</option> option instead.
+        </para>
+        <para>
+        Setting this option will override any configuration file auto-generated
```

Comment:
Shouldn't this be disallowed by assertion? Allowing a `settings` object that's going to have no effect since it's overridden adds potential for confusion but little value, doesn't it?